sql/src/test/java/org/apache/druid/sql/calcite/CalciteWindowQueryTest.java

```diff
@@ -249,6 +250,29 @@ public void testWithArrayConcat()
         .run();
   }
 
+  @Test
+  public void testFailure_partitionByMVD()
+  {
+    final DruidException e = Assert.assertThrows(
+        DruidException.class,
+        () -> testBuilder()
+            .sql("select cityName, countryName, array_to_mv(array[1,length(cityName)]),\n"
+                 + "row_number() over (partition by  array_to_mv(array[1,length(cityName)]) order by countryName, cityName)\n"
+                 + "from wikipedia\n"
+                 + "where countryName in ('Austria', 'Republic of Korea') and cityName is not null\n"
+                 + "order by 1, 2, 3")
+            .queryContext(DEFAULT_QUERY_CONTEXT)
+            .expectedResults(ImmutableList.of())
+            .run()
+    );
+
+    assertEquals(
```

Comment:
Just to confirm, is this a planning layer error?
Asking as we have `assertQueryIsUnplannable()` method available to validate planning layer error messages. So we should use that if it's a planning layer error.

Reply:
No, I could not find a way to distinguish MVDs at planning stage. This is a query runtime exception when we read the actual data, we validate it to see if it has multiple values.
In a way we do this check on reading every row if the column type is String.

Reply:
Thanks for confirming!